### DISCUSSION TITLE
Hiking DDS238-2 WIFI (based on Tuya MCU module) does not support VoltageSet?

### DISCUSSION THREAD

**Discussion by jorge-chavez** [arendst/Tasmota#12450] [Q&A] [answered]
Hi, I configured my "Hiking DDS238-2 WIFI" following [(https://github.com/krikk/Hiking-DDS238-2-WIFI-Din-Rail-Energy-Meter-flashing-Tasmota)] guidelines using the latest "tasmota.bit" firmware 9.5.0, but I found wrong numbers in:
    - Device LCD 
    - Reported Numbers
If no load is connected, it should be: Voltage: 237.2v, Current: 0.0A, Power: 0W, but this is shown..
![image](https://user-images.githubusercontent.com/67099877/123047673-dcd40080-d3fd-11eb-9274-8f8d2b4bdcdf.png)
Also when known load (incandescent light bulb) is used, weird data appears...

I tried to perform calibration with VoltageSet, CurrentSet, but seems "Tuya MCU(54)" (starting point of the previous procedure, before applying the Template), does not allow  calibration procedure...   

I'm so confused:
- I think meter has two internal processors: one "main" for measuring, time-keeping, ... and LCD local monitoring, and a ESP8266 based for external communications. I mean tasmota changes the ESP8266 behavior, but not the one for the "main".. 
- Is there any way to enable the calibration procedure with "Tuya MCU" based module?

Thanks for your help

**Comment by barbudor** [comment, marked as answer]
Can you please try with [9.4.0](http://sidweb.nl/tasmota/release-9.4.0/) ?
There have been some changes between 9.4.0 and 9.5.0 that impact Tuya power measurements.
Thanks.

**Reply by jorge-chavez** [reply]
Hi,
Thanks for your quick answer

The problem persists :-(
Let me detail some of my trials:
    - As the guideline suggest I started with module "Tuya MCU (54)" and then applied Template: {"NAME":"Hiking DDS238-2 WIFI","GPIO":[0,2272,0,2304,0,0,0,0,0,0,320,0,32,0],"FLAG":0,"BASE":54}
    - Also tested module "Generic (18)" and applied the same Template
In both cases reading are bad (LCD and reported) and does not allow calibration:

```
15:24:00.508 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA03070008120200040000000C35","Cmnd":7,"CmndData":"120200040000000C","DpType2Id18":12,"18":{"DpId":18,"DpIdType":2,"DpIdData":"0000000C"}}}
15:24:01.500 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA03070008130200040000003B65","Cmnd":7,"CmndData":"130200040000003B","DpType2Id19":59,"19":{"DpId":19,"DpIdType":2,"DpIdData":"0000003B"}}}
15:24:02.500 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030700081402000400003DB018","Cmnd":7,"CmndData":"1402000400003DB0","DpType2Id20":15792,"20":{"DpId":20,"DpIdType":2,"DpIdData":"00003DB0"}}}
15:24:03.470 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
15:24:03.501 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030700086502000400000074F0","Cmnd":7,"CmndData":"6502000400000074","DpType2Id101":116,"101":{"DpId":101,"DpIdType":2,"DpIdData":"00000074"}}}
15:24:14.432 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
15:24:25.474 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
15:24:29.887 CMD: VoltageSet 236;
15:24:29.894 MQT: stat/tasmota_88D294/RESULT = {"Command":"Error"}
```

- doesn't seem like a scaling problem, I see the evolution of voltage signal compared with another sensor and they didn't resemble each other (ignoring shifts and scales)
- Power calibration  (https://tasmota.github.io/docs/Power-Monitoring-Calibration/) is not available for Tuya TX/RX components?

I am a bit lost, can you please tell me were I'm wrong

Thanks in advance

**Reply by barbudor** [reply]
I think that power calibration is not available for Tuya devices. The message received from the TuyaMCU should be correct in itself (the Tuya part should already be calibrated).

Are you already with `WebLog 3`
If not , could you please et `WebLog3` and send more logs ?

**Reply by jorge-chavez** [reply]
```
16:04:59.592 CMD: WebLog 3
16:04:59.595 SRC: WebConsole from 192.168.1.104
16:04:59.598 CMD: Group 0, Index 1, Command "WEBLOG", Data "3"
16:04:59.606 MQT: stat/tasmota_88D294/RESULT = {"WebLog":3}
16:05:01.129 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA03070008120200040000000C35","Cmnd":7,"CmndData":"120200040000000C","DpType2Id18":12,"18":{"DpId":18,"DpIdType":2,"DpIdData":"0000000C"}}}
16:05:01.132 TYA: fnId=32 is set for dpId=18
16:05:01.135 TYA: RX value 12 from dpId 18 
16:05:01.137 TYA: Rx ID=18 Current=12
16:05:02.129 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA03070008130200040000003A64","Cmnd":7,"CmndData":"130200040000003A","DpType2Id19":58,"19":{"DpId":19,"DpIdType":2,"DpIdData":"0000003A"}}}
16:05:02.132 TYA: fnId=31 is set for dpId=19
16:05:02.135 TYA: RX value 58 from dpId 19 
16:05:02.138 TYA: Rx ID=19 Active_Power=58
16:05:03.129 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030700081402000400003DA40C","Cmnd":7,"CmndData":"1402000400003DA4","DpType2Id20":15780,"20":{"DpId":20,"DpIdType":2,"DpIdData":"00003DA4"}}}
16:05:03.133 TYA: fnId=33 is set for dpId=20
16:05:03.136 TYA: RX value 15780 from dpId 20 
16:05:03.139 TYA: Rx ID=20 Voltage=15780
16:05:03.444 TYA: Send "55aa001c0008011506171005030371"
16:05:04.130 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030700086502000400000075F1","Cmnd":7,"CmndData":"6502000400000075","DpType2Id101":117,"101":{"DpId":101,"DpIdType":2,"DpIdData":"00000075"}}}
16:05:04.134 TYA: fnId=0 is set for dpId=101
16:05:04.136 TYA: RX value 117 from dpId 101 
16:05:09.451 TYA: Send "55aa00000000ff"
16:05:09.464 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
16:05:09.467 TYA: Heartbeat
16:05:20.454 TYA: Send "55aa00000000ff"
16:05:20.471 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
16:05:20.475 TYA: Heartbeat
16:05:31.129 MQT: tele/tasmota_88D294/RESULT = {"TuyaReceived":{"Data":"55AA03070008120200040000000C35","Cmnd":7,"CmndData":"120200040000000C","DpType2Id18":12,"18":{"DpId":18,"DpIdType":2,"DpIdData":"0000000C"}}}
16:05:31.132 TYA: fnId=32 is set for dpId=18
16:05:31.135 TYA: RX value 12 from dpId 18 

```
Is normal that LCD information is also wrong? I mean, as there are other information (as Frequency) not available from Tuya, I assume there was a different microprocessor/firmware, not related with communications...

**Reply by barbudor** [reply]
As far as I can see with the logs, Tasmota gets the values send by Tuya correctly
Now if the values are bad on the LCD (but matches what Tasmota sees), I think you found the reason why 
🗑️

**Reply by jorge-chavez** [reply]
You mean this unit its faulty, or somehow was bricked by me?

**Reply by barbudor** [reply]
The display is not managed by Tasmota
So if the display shows wrong values, I looks like it's buggy
Maybe ask @krikk opinion about it

**Reply by krikk** [reply]
the values reported by tasmota (tuya), always match the values on the LCD, for my unit, if the LCD displays "wrong" values i would suggest its buggy...

**Reply by sfromis** [reply]
You could try reading the manual for the device. With luck, there might be some calibration feature available, outside the scope of Tasmota.

**Reply by jorge-chavez** [reply]
Is there a procedure for a "configuration factory reset"? I tested many things and some of them are kept even if I reflashed firmware, as if there's some configuration eeprom...
The manual of the module is very basic functional one, I found no reference manual about it.
I'll also check it internal solderings are Ok, because the  flashing procedure I had to unsolderlder/so the ESP module....
Thanks to all for your answers

**Reply by barbudor** [reply]
Reflashing only the firmware without a full erase keeps all settings.
Either you erase before flashing or you use one of the variants of the 'reset' command (see doc). Some variants erase all, some everything but wifi/mqtt....

**Reply by jorge-chavez** [reply]
After opening and checking, I found there's one component missing (drop due to cold soldering), I replaced and everything works fine
![image](https://user-images.githubusercontent.com/67099877/123225213-c8156c80-d4d2-11eb-854b-6d5de38d9844.png)
Thanks again for your help

**Reply by barbudor** [reply]
Great news !
